pkg/api/kpm_run_test.go

```diff
@@ -214,3 +215,32 @@ func TestRunWithOptsAndNoSumCheck(t *testing.T) {
 		assert.Equal(t, err, nil)
 	}
 }
+
+func TestRunWithOptsWithNoLog(t *testing.T) {
+	pkgPath := filepath.Join(getTestDir("test_run_pkg_in_path"), "test_run_with_no_log")
+
+	defer func() {
+		_ = os.Remove(filepath.Join(pkgPath, "kcl.mod.lock"))
+	}()
+
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	pathMainK := filepath.Join(pkgPath, "main.k")
+
+	_, err := RunWithOpts(
+		opt.WithEntries([]string{pathMainK}),
+		opt.WithKclOption(kcl.WithWorkDir(pkgPath)),
+		opt.WithLogWriter(nil),
```

Comment:
It is best to add different option orders to test and verify idempotence.

Reply:
fixed